addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -1514,6 +1514,25 @@ def test_message_process_file_encoding_ascii(self):
         # This explains the multiple "�" in the attachment.
         self.assertIn("Chauss������e de Bruxelles", record.message_main_attachment_id.raw.decode())
 
+    def test_message_route_reply_model_none(self):
+        message = self.env['mail.message'].create({
+            'email_from': self.email_from,
+            'body': '<p>test</p>',
+            'message_type': 'email',
+            'model': None,
+            'res_id': None,
+        })
+
+        self.env['mail.alias'].create({'alias_name': 'test', 'alias_model_id': self.env['ir.model']._get('res.partner').id})
+        record = self.format_and_process(
+            MAIL_TEMPLATE, self.email_from, 'test@test.com',
+            subject=message.message_id, extra='In-Reply-To:\r\n\t%s\n' % message.message_id, model=None, target_model='res.partner',
+            target_field='name')
+        self.assertGreater(len(record), 0)
+        self.assertEqual(record._name, 'res.partner')
```

Comment:
Could check it has a message related to the message coming from format_and_process, and that its parent_id is not "message" here above.

Reply:
@tde-banana-odoo hi, can you explain little bit more?

Reply:
I mean you could access ``record.message_ids`` to have access to its message, find the one related to the newly-posted one, and check its ``parent_id`` field. Normally you should have a generic creation message then the email message, with its parent being set to the creation message. You can check the record messages, their subject / parent_id to see a bit what is involved :) .

Purpose of the test is to check the message was posted at the right place with the right hierarchy, so it would makes sense to check everything is ok from that point of view :) .